### DISCUSSION TITLE
Batch Insert into Mysql

### DISCUSSION THREAD

**Discussion by fR0DDY** [playframework/playframework#11563] [Q&A]
I am having trouble doing batch inserts into MySQL using the java play framework through hibernate and JPA. Any pointers?

I have tried :

1. Using entity flush, clear after every x records in a transaction
2. Adding hibernate.jdbc.batch_size=5 property to application.conf 
3. Adding rewriteBatchedStatements=true to DB URL.

**Comment by francisdb** [comment]
Do you have a bit more details on the "trouble" you are having? Exceptions? Stack traces?

**Comment by fR0DDY** [comment]
There is no exception as such but in my debug logs I can see that the queries are still not batched and executed individually.

**Comment by mkurz** [comment]
I am pretty sure putting a hibernate property in Play's `application.conf` wont work.
I bet you need to set that property in the `conf/META-INF/persistence.xml` config, see https://www.playframework.com/documentation/2.8.x/JavaJPA#Creating-a-Persistence-Unit.

**Reply by mkurz** [reply]
I am talking about `hibernate.jdbc.batch_size=5` btw.

**Comment by fR0DDY** [comment]
Tried that as well. Didn't help. Still individual inserts.

**Reply by mkurz** [reply]
Please post your `conf/META-INF/persistence.xml`. I am pretty sure this has nothing to do with Play.

**Reply by fR0DDY** [reply]
```
<persistence xmlns="http://xmlns.jcp.org/xml/ns/persistence"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/persistence http://xmlns.jcp.org/xml/ns/persistence/persistence_2_1.xsd"
  version="2.1">

  <persistence-unit name="com.hydrogen.jpa" transaction-type="RESOURCE_LOCAL">
    <provider>org.hibernate.jpa.HibernatePersistenceProvider</provider>
    <non-jta-data-source>DefaultDS</non-jta-data-source>
    <properties>
      <property name="hibernate.dialect" value="org.hibernate.dialect.MySQL8Dialect"/>
      <property name="hibernate.show_sql" value="true"/>
      <property name="hibernate.jdbc.batch_size" value="5"/>
      <property name="hibernate.hbm2ddl.auto" value="update"/>
    </properties>
  </persistence-unit>

</persistence>
```

**Reply by mkurz** [reply]
I fear I can not help here, IMHO this is a configuration problem on your side. Make sure you read and understand https://www.playframework.com/documentation/2.8.x/JavaJPA. Also for your issue see this answer and the link there on https://stackoverflow.com/a/21259491/810109 - maybe there is some information that helps.
